PrivateClinicsWebNet.Infrastructure/AppointmentInfrastructure/Repositories/AppointmentRepository.cs

```diff
@@ -0,0 +1,79 @@
+﻿using Microsoft.EntityFrameworkCore;
+using PrivateClinicsWebNet.BusinessLogic.Abstractions;
+using PrivateClinicsWebNet.BusinessLogic.Entities;
+using PrivateClinicsWebNet.DataAccess;
+using PrivateClinicsWebNet.Infrastructure.AppointmentInfrastructure.Abstractions;
+using PrivateClinicsWebNet.Infrastructure.Shared.Wrapper;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace PrivateClinicsWebNet.Infrastructure.AppointmentInfrastructure.Repositories
+{
+    public class AppointmentRepository : IAppointmentRepository
+    {
+        private readonly ApplicationDbContext _context;
+
+        public AppointmentRepository(ApplicationDbContext context)
+        {
+            _context = context;
+        }
+
+        public async Task<Appointment> GetAppointmentByIdAsync(string id)
+        {
+            var appointment = await _context.Appointments
+                .Include(entity=>entity.Patient)
+                .Include(entity=>entity.Doctor)
+                .FirstOrDefaultAsync(x => x.Id.ToString() == id);
+            return appointment;
+        }
+
+        public async Task<bool> СreateAppointmentAsync(Appointment appointment)
+        {
+            var result = await _context.Appointments
+                .AddAsync(appointment);
+            await _context.SaveChangesAsync();
+            return result.IsKeySet;
+        }
+
+        public async Task UpdateAppointmentAsync(Appointment appointment)
+        {
+            var existingAppointment = await _context.Appointments.FindAsync(appointment.Id);
+            if(appointment.DoctorId != null)
+                existingAppointment.DoctorId = appointment.DoctorId;
+            if (appointment.PatientId != null)
+                existingAppointment.PatientId = appointment.PatientId;
+            if(appointment.Date != null)
+                existingAppointment.Date = appointment.Date;
+            await _context.SaveChangesAsync();
+        }
+
+        public async Task<List<Appointment>> GetUserAppointmentsAsync(Expression<Func<Appointment, bool>> matchesUserId,
```

Comment:
This violates separation of concerns. We don't pass expressions to repository methods. Just pass parameters andthen create needed expressions on the repo level